src/Components/src/Microsoft.AspNetCore.Components/ParameterCollectionExtensions.cs

```diff
@@ -37,23 +35,36 @@ public static class ParameterCollectionExtensions
             }
 
             var targetType = target.GetType();
-            if (!_cachedParameterWriters.TryGetValue(targetType, out var parameterWriters))
+            if (!_cachedWritersByType.TryGetValue(targetType, out var writers))
             {
-                parameterWriters = CreateParameterWriters(targetType);
-                _cachedParameterWriters[targetType] = parameterWriters;
+                writers = new WritersForType(targetType);
+                _cachedWritersByType[targetType] = writers;
```

Comment:
nit: changing this to `TryAdd` is a little more friendly and makes the pattern obvious.